script/src/syscalls/load_cell_data.rs

```diff
@@ -104,13 +107,15 @@ impl<'a, DL: CellDataProvider + 'a> LoadCellData<'a, DL> {
             .load_cell_data(cell)
             .ok_or(VMError::Unexpected)?
             .slice((content_offset as usize)..(content_end as usize));
-        machine.memory_mut().init_pages(
-            addr,
-            memory_size,
-            FLAG_DIRTY | FLAG_EXECUTABLE | FLAG_FREEZED,
-            Some(data),
-            0,
-        )?;
+        // To be consistent with the mainnet, add this flag after hardfork
```

Comment:
I made a mistake before. Any write operation to the ckb-vm memory will be marked as dirty, and the correct logic should be:

```rs
machine.memory_mut().init_pages(
            addr,
            memory_size,
            FLAG_EXECUTABLE | FLAG_FREEZED,
            Some(data),
            0,
        )?;
        if !self.dirty_flag_enable {
            let mut current_addr = addr;
            while current_addr < addr + memory_size {
                let page = current_addr / RISCV_PAGESIZE as u64;
                machine.memory_mut().clear_flag(page, FLAG_DIRTY)?;
                current_addr += RISCV_PAGESIZE as u64;
            }
        }
```